pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go

```diff
@@ -144,7 +144,8 @@ func (r *HPAReconciler) checkHPAExist(client client.Client) (constants.CheckResu
 }
 
 func semanticHPAEquals(desired, existing *autoscalingv2.HorizontalPodAutoscaler) bool {
-	return equality.Semantic.DeepEqual(desired.Spec.Metrics, existing.Spec.Metrics) &&
+	return equality.Semantic.DeepEqual(desired.Spec.ScaleTargetRef, existing.Spec.ScaleTargetRef) &&
```

Comment:
Good catch!
Then, why don't we add  `Behavior` too? 
```suggestion
	return equality.Semantic.DeepEqual(desired.Spec.ScaleTargetRef, existing.Spec.ScaleTargetRef) &&
	equality.Semantic.DeepEqual(desired.Spec.Behavior, existing.Spec.Behavior) &&
```

Reply:
Then this is simply the entire autoscaler spec.